app/middleware/authenticator.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+require './app/models/user'
+
+def logged_in?
+  !session[:user_id].nil?
+end
+
+# Authenticate user.
+def authenticate
+  if logged_in?
+    @user = User.find_by_id(session[:user_id])
+    @admin = @user.admin?
```

Comment:
it would be good to separate these. Like have an admin user separate from a normal user